config/initializers/commontator.rb

```diff
@@ -1,187 +1,249 @@
-
-  # Whether users automatically subscribe to a thread when commenting
-  # Default: false
-  config.auto_subscribe_on_comment = false
-
-  # Whether users can edit their own comments
-  # Default: true
-  config.can_edit_own_comments = false # REVIEW: Temporarily disabled
+  # User (acts_as_commontator) Configuration
 
+  # user_name_proc
+  # Type: Proc
+  # Arguments: a user (acts_as_commontator)
+  # Returns: the user's name (String)
+  # Default: lambda { |user| I18n.t('commontator.anonymous') } (all users are anonymous)
+  config.user_name_proc = lambda { |user| I18n.t('commontator.anonymous') }
+
+  # user_link_proc
+  # Type: Proc
+  # Arguments: a user (acts_as_commontator),
+  #            the app_routes (ActionDispatch::Routing::RoutesProxy)
+  # Returns: a path to the user's `show` page (String)
+  # If anything non-blank is returned, the user's name in comments
+  # comments will become a hyperlink pointing to this path
+  # The main application's routes can be accessed through the app_routes object
+  # Default: lambda { |user, app_routes| '' } (no link)
+  config.user_link_proc = lambda { |user, app_routes| '' }
+
+  # user_avatar_proc
+  # Type: Proc
+  # Arguments: a user (acts_as_commontator), a view (ActionView::Base)
+  # Returns: a String containing a HTML <img> tag pointing to the user's avatar image
+  # The commontator_gravatar_image_tag helper takes a user object,
+  # a border size and an options hash for Gravatar, and produces a Gravatar image tag
+  # See available options at http://en.gravatar.com/site/implement/images/)
+  # Note: Gravatar has several security implications for your users
+  #       It makes your users trackable across different sites and
+  #       allows de-anonymization attacks against their email addresses
+  #       If you absolutely want to keep users' email addresses or identities secret,
+  #       do not use Gravatar or similar services
+  # Default: lambda { |user, view|
+  #            view.commontator_gravatar_image_tag(
+  #              user, 1, :s => 60, :d => 'mm') }
+  config.user_avatar_proc = lambda { |user, view|
+                                     view.commontator_gravatar_image_tag(
+                                       user, 1, :s => 60, :d => 'mm') }
+
+  # user_email_proc
+  # Type: Proc
+  # Arguments: a user (acts_as_commontator), a mailer (ActionMailer::Base)
+  # Returns: the user's email address (String)
+  # The default works for Devise's defaults
+  # If the mailer argument is nil, Commontator intends to hash the email and send the hash
+  # to Gravatar, so you should always return the user's email address (if using Gravatar)
+  # If the mailer argument is not nil, then Commontator intends to send an email to
+  # the address returned; you can prevent it from being sent by returning a blank String
+  # Default: lambda { |user, mailer| user.try(:email) || '' }
+  config.user_email_proc = lambda { |user, mailer| user.try(:email) || '' }
```

Comment:
Unused block argument - `mailer`. If it's necessary, use `_` or `_mailer` as an argument name to indicate that it won't be used.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.